diff --git i/src/uct/ze/copy/ze_copy_iface.c w/src/uct/ze/copy/ze_copy_iface.c
index f09949933..3a7e392fb 100644
--- i/src/uct/ze/copy/ze_copy_iface.c
+++ w/src/uct/ze/copy/ze_copy_iface.c
@@ -193,12 +193,22 @@ uct_ze_copy_estimate_perf(uct_iface_h tl_iface, uct_perf_attr_t *perf_attr)
     return UCS_OK;
 }
 
+static void stub_function_empty_function(struct uct_iface *) {}
+static ucs_status_t stub_function_return_unsupported_1(struct uct_ep *, uct_ep_attr_t *)
+{
+    return UCS_ERR_UNSUPPORTED;
+}
+static ucs_status_t stub_function_return_unsupported_2(struct uct_ep *, unsigned int)
+{
+    return UCS_ERR_UNSUPPORTED;
+}
+
 
 static uct_iface_internal_ops_t uct_ze_copy_iface_internal_ops = {
     .iface_estimate_perf   = uct_ze_copy_estimate_perf,
-    .iface_vfs_refresh     = ucs_empty_function,
-    .ep_query              = ucs_empty_function_return_unsupported,
-    .ep_invalidate         = ucs_empty_function_return_unsupported,
+    .iface_vfs_refresh     = stub_function_empty_function,
+    .ep_query              = stub_function_return_unsupported_1,
+    .ep_invalidate         = stub_function_return_unsupported_2,
     .ep_connect_to_ep_v2   = (uct_ep_connect_to_ep_v2_func_t)ucs_empty_function_return_unsupported,
     .iface_is_reachable_v2 = uct_ze_copy_iface_is_reachable_v2,
     .ep_is_connected       = uct_base_ep_is_connected
diff --git i/src/uct/ze/copy/ze_copy_md.c w/src/uct/ze/copy/ze_copy_md.c
index f60ec1ebd..9750958f2 100644
--- i/src/uct/ze/copy/ze_copy_md.c
+++ w/src/uct/ze/copy/ze_copy_md.c
@@ -241,6 +241,15 @@ uct_ze_copy_md_detect_memory_type(uct_md_h md, const void *addr, size_t length,
     return UCS_OK;
 }
 
+static ucs_status_t stub_function_return_success(struct uct_md *, const uct_md_mem_dereg_params_t *)
+{
+    return UCS_OK;
+}
+static ucs_status_t stub_function_return_success_2(struct uct_md *, void *, void *, size_t,  const uct_md_mkey_pack_params_t *, void *)
+{
+    return UCS_OK;
+}
+
 static uct_md_ops_t md_ops = {
     .close              = uct_ze_copy_md_close,
     .query              = uct_ze_copy_md_query,
@@ -248,9 +257,9 @@ static uct_md_ops_t md_ops = {
     .mem_free           = uct_ze_copy_mem_free,
     .mem_advise         = (uct_md_mem_advise_func_t)ucs_empty_function_return_unsupported,
     .mem_reg            = uct_ze_copy_mem_reg,
-    .mem_dereg          = ucs_empty_function_return_success,
+    .mem_dereg          = stub_function_return_success,
     .mem_query          = uct_ze_copy_md_mem_query,
-    .mkey_pack          = ucs_empty_function_return_success,
+    .mkey_pack          = stub_function_return_success_2,
     .mem_attach         = (uct_md_mem_attach_func_t)ucs_empty_function_return_unsupported,
     .detect_memory_type = uct_ze_copy_md_detect_memory_type,
 };
